pkg/gui/controllers/branches_controller.go

```diff
@@ -234,6 +243,23 @@ func (self *BranchesController) setUpstream(selectedBranch *models.Branch) error
 		Key:     'g',
 	}
 
+	upstreamRebaseItem := &types.MenuItem{
+		LabelColumns: []string{upstreamRebaseOptions},
+		OpensMenu:    true,
+		OnPress: func() error {
+			if err := self.c.Helpers().MergeAndRebase.RebaseOntoRef(selectedBranch.ShortUpstreamRefName()); err != nil {
+				return self.c.Error(err)
+			}
+			return nil
+		},
+		Tooltip: upstreamRebaseTooltip,
+		Key:     'r',
+	}
+
+	if !selectedBranch.IsTrackingRemote() || selectedBranch.UpstreamGone {
```

Comment:
Seems to me we should be able to use `if !selectedBranch.RemoteBranchStoredLocally()` like we do for the others below.